legal-api/src/legal_api/core/meta/filing.py

```diff
@@ -337,3 +339,7 @@ def alter_outputs(filing_type: str, filing_meta_data: dict, outputs: set):
                 # legalFilings list contains correction and incorporationApplication
                 # and it should be removed if correction does not contain name change.
                 outputs.remove('certificate')
+            if filing.get('filing', {}).get('business', {}).get('legalType') == 'CP' and \
+                    'annualReport' in filing_meta_data.get('legalFilings', []) and 'noticeOfArticles' in outputs:
```

Comment:
i don't think you will be able to check for "annualReport" in `filing_meta_data` here.  From looking at the `filing.meta_data` field for filings in Dev and Prod, `annualReport` is not included in the `legalFilings` property.  I do see that for other corrections like an incorporationApplication its included.  Maybe this is a bug?  do you have any insight into this @vysakh-menon-aot @lekshmimallika-aot ?

Meta data in prod for AR correction filings.
![image](https://user-images.githubusercontent.com/6685223/176209267-b5864bd2-58cd-42c0-9a27-48d5049c834a.png)

We typically don't want to reference the `filing.filing_json` when we don't have to but we can check if the `correctedFilingType` is "annualReport" in the correction filing.  If we don't do this, we may have to do a datafix and potentially update the correction logic for ARs if it is the case that "annualReport" should be included in the "legalFilings" array in the filing meta data.

Reply:
I was actually talking about using the `correctedFilingType` field in the `filing_json` to determine if the correction filing is for an AR.  

![image](https://user-images.githubusercontent.com/6685223/176223635-1d0ef3b9-8864-4b2d-a801-169b82e76436.png)

The `filing_json` field can be found in the `filings` model.  

![image](https://user-images.githubusercontent.com/6685223/176224075-8b17b4ec-454a-4ca0-ba62-b3ea54068352.png)

But lets wait to see if @vysakh-menon-aot or @lekshmimallika-aot has any other thoughts first.  And whether what I mentioned in my initial comment about "annualReport" not being in the `legalFilings` array in the `filing.meta_data` is a bug.

Reply:
In that case I think your original code of looking for "annualReport" in the `legalFilings` array would work @lewischenstudio .  We just need to do a data fix in prod and create a ticket to fix the AR correction bug to populate "annualReport" in `legalFilings` in the `filings.meta_data` field.

Reply:
@lewischenstudio We should do the datafix in prod and go with your initial solution as per my previous comment.  The filing_json structure is not guaranteed to be always the same so we want to avoid using it directly whenever possible.  

But before all of this, can you confirm with Linda regarding the question above from @lekshmimallika-aot?  about whether we even need any NOAs at all for CPs?  if not, then it sounds like we don't even need this check?

Reply:
@argush3 @lekshmimallika-aot I've got an update from Linda. It is right that we want to suppress NOA from all COOP records if they belong to type "correction" and "annual report" filing.

Reply:
Ok, I'll write the script to update data in prod and all the other environments.  I'll let you know when the data has been updated @lewischenstudio in the ticket.  

I think I'm good with what you have currently but can we pass the business legal type as a parameter for `alter_outputs` function instead of retrieving the business again?

Reply:
Thanks Argus. Yes, I can pass the business legal type as a parameter for `alter_outputs`, but no other place is accessing `business` than the inside the condition `filing_type == 'correction'. If I declare this outside `alter_ouputs` and end up not using it in the if statement, then it's an extra step that's not used. That's why I use this only for the condition `filing_type == 'correction'`. 

Let me know if you still want to pass this parameter to `alter_outputs`, and I'll update the method.

Reply:
yes, please pass the `legal_type` as a parameter for `alter_outputs`.  I understand what you are saying but it's not that much work to remove legal_type if we don't end up using it.